### PR TITLE
Raise an exception if check_val_every_n_epoch is not an integer

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Raise an exception if check_val_every_n_epoch is not an integer

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/data_connector.py [hunk 1]
```diff
@@ -30,6 +30,12 @@ def on_trainer_init(self, check_val_every_n_epoch, reload_dataloaders_every_epoc
         self.trainer.datamodule = None
```
Comment (ananthsub) [inline pytorch_lightning/trainer/connectors/data_connector.py:30]: could you add typehints for the arguments in `on_trainer_init`?

Comment (kaushikb11) [reply pytorch_lightning/trainer/connectors/data_connector.py:32]: Done!

**Comment by ananthsub** [comment]
What's the issue this solves? `check_val_every_n_epoch` is typed as an `int` already: https://pytorch-lightning.readthedocs.io/en/latest/common/trainer.html#init

**Comment by kaushikb11** [comment]
@ananthsub It's about the expected behavior. It also works for `float`.

Example, for `check_val_every_n_epoch=1.5`, it will check val for 3rd, 6th, .. epoch.

I don't believe it's in alignment with what's mentioned in the docs and expected. Also, I think if the User is under an impression that it supports float as well, then they would expect `int(trainer.num_training_batches * check_val_every_n_epoch)` to check val.

Related [Code](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/training_loop.py#L826).

File: pytorch_lightning/trainer/connectors/data_connector.py [hunk 2]
```diff
@@ -26,10 +26,18 @@ class DataConnector(object):
     def __init__(self, trainer):
         self.trainer = trainer
 
-    def on_trainer_init(self, check_val_every_n_epoch, reload_dataloaders_every_epoch, prepare_data_per_node):
+    def on_trainer_init(
+        self, check_val_every_n_epoch: int, reload_dataloaders_every_epoch: bool, prepare_data_per_node: bool
+    ):
```
Comment (ananthsub) [inline pytorch_lightning/trainer/connectors/data_connector.py:31]:
```suggestion
    ) -> None:
```

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/trainer/connectors/data_connector.py [hunk 3]
```diff
@@ -26,10 +26,18 @@ class DataConnector(object):
...
+    def on_trainer_init(
+        self, check_val_every_n_epoch: int, reload_dataloaders_every_epoch: bool, prepare_data_per_node: bool
+    ) -> None:
         self.trainer.datamodule = None
         self.trainer.prepare_data_per_node = prepare_data_per_node
 
+        if not isinstance(check_val_every_n_epoch, int):
+            raise MisconfigurationException(
+                f"check_val_every_n_epoch={check_val_every_n_epoch} is not a"
+                " valid configuration. It should be an integer."
```
Comment (tchaton) [inline pytorch_lightning/trainer/connectors/data_connector.py:38]:
```suggestion
                f"check_val_every_n_epoch should be an integer. Found {check_val_every_n_epoch}"
```

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by carmocca — APPROVED** [review]

**Pull request merged**